benchmark/cublas/blas3/gemm_batched_strided.cpp

```diff
@@ -0,0 +1,240 @@
+/**************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename gemm_batched_strided.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string t1, std::string t2, int m, int k, int n,
+                     int batch_size) {
+  std::ostringstream str{};
+  str << "BM_GemmBatchedStrided<"
+      << blas_benchmark::utils::get_type_name<scalar_t>() << ">/" << t1 << "/"
+      << t2 << "/" << m << "/" << k << "/" << n << "/" << batch_size << "/";
+
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void cublas_routine(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CUBLAS_CHECK(cublasSgemmStridedBatched(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CUBLAS_CHECK(cublasDgemmStridedBatched(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr, int t1,
+         int t2, index_t m, index_t k, index_t n, scalar_t alpha, scalar_t beta,
```

Comment:
You can look at the corresponding unit-tests for reference : https://github.com/codeplaysoftware/sycl-blas/blob/master/test/unittest/blas3/blas3_gemm_common.hpp#L187

Reply:
Thank you @OuadiElfarouki for your review.
This benchmark is the counterpart of the same one for syclblas which is currently broken. I implemented it because cuBLAS uses two APIs that store matrices differently and I wanted to benchmark both. I will look at the example you sent and check what could be done.